protos/flyteidl/admin/matchable_resource.proto

```diff
@@ -54,7 +54,13 @@ message TaskResourceSpec {
 message TaskResourceAttributes {
   TaskResourceSpec defaults = 1;
 
+  // These are limits from the system's perspective. Resources requested by the user (either in defaults or limits)
+  // are checked against these limits. Defaults in this message, (fields 1 and 3) are also checked against these limits.
   TaskResourceSpec limits = 2;
+
+  // If set, these limits will be used as the default limits for tasks that do not specify limits. If not set, but
+  // defaults (field 1) are set, then the defaults will be used as the limits.
+  TaskResourceSpec default_limits = 3;
```

Comment:
Because this naming has tripped people up somewhat already, what if we differentiated between defaults and limits like so

```
message TaskResourceAttributes
   TaskResourceSpec defaults = 1 [deprecated=true];

    TaskResourceSpec limits = 2;

   
   TaskResourceDefaults default_values = 1;
}

message TaskResourceDefaults{
  TaskResourceSpec requests = 1;

  TaskResourceSpec limits = 2;
}
```

So it's clear exactly what is a default and what is a limit

Reply:
+1 to changing this terminology somehow. Also, didn't realize until now that changing the name of a proto message is easy if you use the same id number! Can we maybe specify something like:
```
message TaskResourceAttributes {
    TaskResourceSpec defaults = 1; [deprecated=true]
    TaskResourceSpec default_requests = 1;
    
    TaskResourceSpec limits = 2; // IIUC these are the hard limits imposed by the platform
    
    TaskResourceSpec default_limits = 3;
}
```